components/application-broker/internal/broker/provision.go

```diff
@@ -394,6 +425,53 @@ func (svc *ProvisionService) channelForApp(applicationName internal.ApplicationN
 	return svc.knClient.GetChannelByLabels(integrationNamespace, labels)
 }
 
+// Create a new policy
+func (svc *ProvisionService) createIstioPolicy(ns internal.Namespace) error {
+	policyName := fmt.Sprintf("%s%s", ns, policyNameSuffix)
+	svc.log.Infof("Creating Policy %s in namespace: %s", policyName, string(ns))
+
+	brokerTargetSelector := &istioauthenticationalpha1.TargetSelector{
+		Name: brokerTargetSelectorName,
+	}
+	filterTargetSelector := &istioauthenticationalpha1.TargetSelector{
+		Name: filterTargetSelectorName,
+	}
+	mtls := &istioauthenticationalpha1.MutualTls{
+		Mode: istioMtlsPermissiveMode,
+	}
+	peerAuthenticationMethod := &istioauthenticationalpha1.PeerAuthenticationMethod{
+		Params: &istioauthenticationalpha1.PeerAuthenticationMethod_Mtls{
+			Mtls: mtls,
+		},
+	}
+
+	policy := &istiov1alpha1.Policy{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      policyName,
+			Namespace: string(ns),
+		},
+		Spec: istioauthenticationalpha1.Policy{
+			Targets: []*istioauthenticationalpha1.TargetSelector{brokerTargetSelector, filterTargetSelector},
+			Peers:   []*istioauthenticationalpha1.PeerAuthenticationMethod{peerAuthenticationMethod},
+		},
+	}
+
+	_, createError := svc.istioClient.AuthenticationV1alpha1().Policies(string(ns)).Create(policy)
+	if createError != nil {
+		if apiErrors.IsAlreadyExists(createError) {
+			if _, updateError := svc.istioClient.AuthenticationV1alpha1().Policies(string(ns)).Update(policy); updateError != nil {
+				svc.log.Printf("Updating Policies %s in namespace: %s failed with createError:\n %s", policyName, ns,
```

Comment:
logging an error here is redundant, because if it occurs you first log it and then return it, and then log the returned error again in line 259. Instead you can wrap it with a message informing about the context of the error occurance:
```suggestion
				return errors.Wrapf(err, "while creating policies %s in namespace: %s failed with error:\n %s", policyName, ns)
```